Swift 5.5.2/Xcode 13.2 refuses to compile this piece as it is "unable to infer complex closure return type."

diff --git a/Sources/_RegexParser/Utility/TypeConstruction.swift b/Sources/_RegexParser/Utility/TypeConstruction.swift
index 4d1765e..39b4595 100644
--- a/Sources/_RegexParser/Utility/TypeConstruction.swift
+++ b/Sources/_RegexParser/Utility/TypeConstruction.swift
@@ -60,7 +60,7 @@ public enum TypeConstruction {
       flags |= 0x10000
     }
     
-    let result = elementTypes.withContiguousStorageIfAvailable { elementTypesBuffer in
+    let result = elementTypes.withContiguousStorageIfAvailable { elementTypesBuffer -> (value: Any.Type, state: Int) in
       if let labels = labels {
         return labels.withCString { labelsPtr in
           swift_getTupleTypeMetadata(
